Classes/Command/ValidatorCommand.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package stefanfroemken/ext-kickstarter.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace StefanFroemken\ExtKickstarter\Command;
+
+use StefanFroemken\ExtKickstarter\Enums\ValidatorType;
+use StefanFroemken\ExtKickstarter\Information\ExtensionInformation;
+use StefanFroemken\ExtKickstarter\Information\ValidatorInformation;
+use StefanFroemken\ExtKickstarter\Service\Creator\ValidatorCreatorService;
+use StefanFroemken\ExtKickstarter\Traits\AskForExtensionKeyTrait;
+use StefanFroemken\ExtKickstarter\Traits\ExtensionInformationTrait;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class ValidatorCommand extends Command
+{
+    use AskForExtensionKeyTrait;
+    use ExtensionInformationTrait;
+
+    public function __construct(
+        private readonly ValidatorCreatorService $validatorCreatorService,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument(
+            'extension_key',
+            InputArgument::OPTIONAL,
+            'Provide the extension key you want to extend.',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Welcome to the TYPO3 Extension Builder');
+
+        $io->text([
+            'We are here to assist you in creating a new TYPO3 validator. ',
+            'Now, we will ask you a few questions to customize the validator according to your needs.',
+            'Please take your time to answer them.',
+            'See https://docs.typo3.org/permalink/t3coreapi:extbase-domain-validator on how implement its functionality.',
+        ]);
+
+        $this->validatorCreatorService->create($this->askForValidatorInformation($io, $input));
+
+        return Command::SUCCESS;
+    }
+
+    private function askForValidatorInformation(SymfonyStyle $io, InputInterface $input): ValidatorInformation
+    {
+        $extensionInformation = $this->getExtensionInformation(
+            $this->askForExtensionKey($io, $input->getArgument('extension_key')),
+            $io
+        );
+
+        $name = $this->askForValidatorName($io);
```

Comment:
$name and $type does not have special handling. You can move the calls direct in new Validator arguments

Reply:
The type does have a scpecial handling, we only ask for the model to use for type model